chcemvediet/apps/inforequests/tests/test_models/test_action.py

```diff
@@ -0,0 +1,874 @@
+                (Action.TYPES.CLARIFICATION_RESPONSE, naive_date(u'2010-10-15'), Deadline.TYPES.OBLIGEE_DEADLINE, dict()),  # 8 WD since delivered_date
+                (Action.TYPES.APPEAL,                 naive_date(u'2010-10-20'), Deadline.TYPES.OBLIGEE_DEADLINE, dict()),  # 15 CD since delivered_date
+                (Action.TYPES.CONFIRMATION,           naive_date(u'2010-10-15'), Deadline.TYPES.OBLIGEE_DEADLINE, dict()),  # previous action deadline
+                (Action.TYPES.EXTENSION,              naive_date(u'2010-10-19'), Deadline.TYPES.OBLIGEE_DEADLINE, dict()),  # previous action deadline + extension
+                (Action.TYPES.ADVANCEMENT,            None, None, dict()),
+                (Action.TYPES.CLARIFICATION_REQUEST,  naive_date(u'2010-10-12'), Deadline.TYPES.APPLICANT_DEADLINE, dict()),  # 7 CD since delivered date
+                (Action.TYPES.DISCLOSURE,             naive_date(u'2010-10-20'), Deadline.TYPES.APPLICANT_DEADLINE, dict(disclosure_level=Action.DISCLOSURE_LEVELS.NONE)),  # 15 CD since delivered_date
+                (Action.TYPES.DISCLOSURE,             naive_date(u'2010-10-20'), Deadline.TYPES.APPLICANT_DEADLINE, dict(disclosure_level=Action.DISCLOSURE_LEVELS.PARTIAL)),  # 15 CD since delivered_date
+                (Action.TYPES.DISCLOSURE,             None, None, dict(disclosure_level=Action.DISCLOSURE_LEVELS.FULL)),
+                (Action.TYPES.REFUSAL,                naive_date(u'2010-10-20'), Deadline.TYPES.APPLICANT_DEADLINE, dict()),  # 15 CD since delivered_date
+                (Action.TYPES.AFFIRMATION,            None, None, dict()),
+                (Action.TYPES.REVERSION,              None, None, dict()),
+                (Action.TYPES.REMANDMENT,             naive_date(u'2010-10-20'), Deadline.TYPES.OBLIGEE_DEADLINE, dict()),  # 8 WD since legal_date advanced by 4 WD
+                (Action.TYPES.ADVANCED_REQUEST,       naive_date(u'2010-10-20'), Deadline.TYPES.OBLIGEE_DEADLINE, dict()),  # 8 WD since legal_date advanced by 4 WD
+                (Action.TYPES.EXPIRATION,             naive_date(u'2010-10-19'), Deadline.TYPES.APPLICANT_DEADLINE, dict()),  # 15 CD since legal_date
+                (Action.TYPES.APPEAL_EXPIRATION,      None, None, dict()),
+                )
+        # Make sure we are testing all defined action types
+        tested_action_types = set(a for a, _, _, _ in tests)
+        defined_action_types = Action.TYPES._inverse.keys()
+        self.assertItemsEqual(tested_action_types, defined_action_types)
+
+        for action_type, deadline_date, deadline_type, extra_args in tests:
+            previous = self._create_action(type=Action.TYPES.REQUEST, delivered_date=delivered_date)  # Deadline for CONFIRMATION, EXTENSION
+            action = self._create_action(type=action_type, delivered_date=delivered_date, legal_date=legal_date, extension=2, **extra_args)
+            dd = action.deadline and action.deadline.deadline_date
```

Comment:
@martinmacko47 v kode sa takymto sposobom vyuziva `or` ale nevsimol som si pouzitie `and`. mozem to pouzit takto, alebo pouzijem `action.deadline.deadline_date if action.deadline else None`

Reply:
@viliambalaz Ano, kludne `and` takto pouzivaj, je to v pythone bezne. Akurat nevidim dovod, naco su ti premenne `dd` a `dt`, preco nie rovno `self.assertEqual(action.deadline and action.deadline.deadline_date, deadline_date)`

Reply:
:ok: